Added Angle Check

leiflang opened the pull request:
Added a thruster angle check, now thrusters wont fire until the rotor is facing the correct direction. it uses the Angle between cos method of obtaining the angle so 1 is correct and the closer to 0 you get the more of an angle difference there is between current angle and desired angle. I plan to implement some sort of linear equation so that thrust isn’t On or Off as it is now. There are a number of ways to do it, including just multiplying the “angle” and thrust but that could cause the thrusters to loose some of its maximum thrust as the angle will almost if ever be exactly 1.

leiflang committed:
fixed constant missing variable type and used setFromvec insted

the constant on line 900 was missing a variable type, I assume it was an int. there is no setFromVecNew in the Rotor variable yet

leiflang committed:
Fixed multi cockpit issue

The original method for detecting multiple cockpits was broken, this verson should be revamped but it works.

leiflang committed:
Reimplomented max RPM

I re implomented the max RPM for the vector based rotation method

leiflang committed:
Added a angle

The scrypt now checks if the thrusters are facing the correct way befor applying thrust. Currently it is binary checking so thrust is On or Off, but this should be modified to be more like a porabla formula so thrust will be applied even when the trusters arent fully lined up. However things are MUCH smoother now for large ships.

1wsx10 reviewed (changes requested):
are you using the ingame editor? do yourself a favor and use something decent.

im not going to accept this because there is a space character on the end of every line. this is something the ingame editor does, and it makes it really hard to read the diffs.

any editor worth its salt will have an option to remove spaces on the end of every line, so you could run it through one of those, then redo the pull request to keep your changes.

you could also run it through sed if you are using linux or git bash or something... eg: `sed -i 's,\(^.*\) $,\1,g' VectorThrust.cs`

leiflang committed:
Fix trailing spaces

I used a combo of Space engineers in game editor Notpad++. I used the trim tail spaces command in notpad++. Also fixed the default settings for rotor speed and accuracy.

leiflang committed:
Fixed some spelling and grammar

As the title says, also removed redundant unused functions

leiflang committed:
Removed redundent lines

leiflang committed:
Multiplied offset with thrust

leiflang commented:
I multiplied the offset Angle with the thrust, after some testing the rotors can get an accuracy of 0.99999, so I didn't bother using a constant to correct for the small error that was created, I also tested it on a planet with a large ship and it didn't fall so it works.

1wsx10 commented:
i manually merged it so i could clean up all the guff with trailing spaces in the diffs.
